Update dependencies

After doing the `sbt run` and loading up http://localhost:9000 - the example app generates an internal error exception.

```
[warn] /Users/eric/Work/scalajs-angular-todomvc/js/src/main/scala/todomvc/example/TodoApp.scala:9: trait JSApp in package js is deprecated: Consult the Scaladoc of js.JSApp for migration tips.
[warn] object TodoApp extends JSApp {
[warn]                        ^
[warn] one warning found
[trace] Stack trace suppressed: run last todomvcJS/compile:scalaJSIR for the full output.
[error] (todomvcJS/compile:scalaJSIR) org.scalajs.core.ir.IRVersionNotSupportedException: Failed to deserialize info of file compiled with Scala.js 0.6.17 (supported: 0.6.0, 0.6.13, 0.6.3, 0.6.4, 0.6.6, 0.6.14, 0.6.8, 0.6.5, 0.6.15): /Users/eric/.ivy2/cache/com.greencatsoft/scalajs-angular_sjs0.6_2.11/jars/scalajs-angular_sjs0.6_2.11-0.8-SNAPSHOT.jar:com/greencatsoft/angularjs/AttributeDirective.sjsir
[trace] Stack trace suppressed: run last todomvcJS/compile:scalaJSIR for the full output.
[error] (todomvcJS/compile:scalaJSIR) org.scalajs.core.ir.IRVersionNotSupportedException: Failed to deserialize info of file compiled with Scala.js 0.6.17 (supported: 0.6.0, 0.6.13, 0.6.3, 0.6.4, 0.6.6, 0.6.14, 0.6.8, 0.6.5, 0.6.15): /Users/eric/.ivy2/cache/com.greencatsoft/scalajs-angular_sjs0.6_2.11/jars/scalajs-angular_sjs0.6_2.11-0.8-SNAPSHOT.jar:com/greencatsoft/angularjs/AttributeDirective.sjsir
[error] application -

! @798b7o079 - Internal server error, for (GET) [/] ->

play.sbt.PlayExceptions$UnexpectedException: Unexpected exception[IRVersionNotSupportedException: Failed to deserialize info of file compiled with Scala.js 0.6.17 (supported: 0.6.0, 0.6.13, 0.6.3, 0.6.4, 0.6.6, 0.6.14, 0.6.8, 0.6.5, 0.6.15): /Users/eric/.ivy2/cache/com.greencatsoft/scalajs-angular_sjs0.6_2.11/jars/scalajs-angular_sjs0.6_2.11-0.8-SNAPSHOT.jar:com/greencatsoft/angularjs/AttributeDirective.sjsir]
	at play.sbt.run.PlayReload$$anonfun$taskFailureHandler$1.apply(PlayReload.scala:51)
	at play.sbt.run.PlayReload$$anonfun$taskFailureHandler$1.apply(PlayReload.scala:44)
	at scala.Option.map(Option.scala:145)
	at play.sbt.run.PlayReload$.taskFailureHandler(PlayReload.scala:44)
	at play.sbt.run.PlayReload$.compileFailure(PlayReload.scala:40)
	at play.sbt.run.PlayReload$$anonfun$compile$2$$anonfun$apply$3.apply(PlayReload.scala:20)
	at play.sbt.run.PlayReload$$anonfun$compile$2$$anonfun$apply$3.apply(PlayReload.scala:20)
	at scala.util.Either$LeftProjection.map(Either.scala:377)
	at play.sbt.run.PlayReload$$anonfun$compile$2.apply(PlayReload.scala:20)
	at play.sbt.run.PlayReload$$anonfun$compile$2.apply(PlayReload.scala:18)
Caused by: org.scalajs.core.ir.IRVersionNotSupportedException: Failed to deserialize info of file compiled with Scala.js 0.6.17 (supported: 0.6.0, 0.6.13, 0.6.3, 0.6.4, 0.6.6, 0.6.14, 0.6.8, 0.6.5, 0.6.15): /Users/eric/.ivy2/cache/com.greencatsoft/scalajs-angular_sjs0.6_2.11/jars/scalajs-angular_sjs0.6_2.11-0.8-SNAPSHOT.jar:com/greencatsoft/angularjs/AttributeDirective.sjsir
	at org.scalajs.core.tools.io.VirtualSerializedScalaJSIRFile$class.info(VirtualFiles.scala:148)
	at org.scalajs.core.tools.io.MemVirtualSerializedScalaJSIRFile.info(MemFiles.scala:104)
	at org.scalajs.core.tools.io.IRFileCache$PersistentIRFile.<init>(IRFileCache.scala:246)
	at org.scalajs.core.tools.io.IRFileCache$PersistedFiles$$anonfun$update$4.apply(IRFileCache.scala:221)
	at org.scalajs.core.tools.io.IRFileCache$PersistedFiles$$anonfun$update$4.apply(IRFileCache.scala:221)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.immutable.List.foreach(List.scala:318)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractTraversable.map(Traversable.scala:105)
Caused by: org.scalajs.core.ir.IRVersionNotSupportedException: This version (0.6.17) of Scala.js IR is not supported. Supported versions are: 0.6.0, 0.6.13, 0.6.3, 0.6.4, 0.6.6, 0.6.14, 0.6.8, 0.6.5, 0.6.15
	at org.scalajs.core.ir.InfoSerializers$Deserializer.readHeader(InfoSerializers.scala:172)
	at org.scalajs.core.ir.InfoSerializers$Deserializer.deserialize(InfoSerializers.scala:103)
	at org.scalajs.core.ir.InfoSerializers$.deserializeWithVersion(InfoSerializers.scala:35)
	at org.scalajs.core.ir.InfoSerializers$.deserialize(InfoSerializers.scala:31)
	at org.scalajs.core.tools.io.VirtualSerializedScalaJSIRFile$class.info(VirtualFiles.scala:145)
	at org.scalajs.core.tools.io.MemVirtualSerializedScalaJSIRFile.info(MemFiles.scala:104)
	at org.scalajs.core.tools.io.IRFileCache$PersistentIRFile.<init>(IRFileCache.scala:246)
	at org.scalajs.core.tools.io.IRFileCache$PersistedFiles$$anonfun$update$4.apply(IRFileCache.scala:221)
	at org.scalajs.core.tools.io.IRFileCache$PersistedFiles$$anonfun$update$4.apply(IRFileCache.scala:221)

```

--- a/build.sbt
+++ b/build.sbt
@@ -1,3 +1,5 @@
+import sbtcrossproject.CrossPlugin.autoImport.{crossProject, CrossType}
+
 organization in ThisBuild := "com.greencatsoft"
 
 version in ThisBuild := "0.8-SNAPSHOT"
@@ -55,7 +57,9 @@ lazy val client = todomvc.js
     skip in packageJSDependencies := false,
     jsEnv in Test := PhantomJSEnv(args = Seq("--web-security=no")).value)
 
-lazy val todomvc = (crossProject in file("."))
+lazy val todomvc = crossProject(JSPlatform, JVMPlatform)
+  .crossType(CrossType.Full)
+  .in(file("."))
   .settings(
     name := "todomvc-common",
     unmanagedSourceDirectories in Compile :=

--- a/build.sbt
+++ b/build.sbt
@@ -16,7 +16,7 @@ lazy val root = project.in(file("."))
   .aggregate(client, server)
   .settings(
     name := "todomvc",
-    run in Compile <<= (run in Compile in server))
+    run in Compile := (run in Compile in server).evaluated)
 
 lazy val server = todomvc.jvm
   .enablePlugins(PlayScala)
@@ -27,7 +27,7 @@ lazy val server = todomvc.jvm
     },
     pipelineStages := Seq(scalaJSProd),
     scalaJSProjects := Seq(todomvc.js),
-    stage <<= stage dependsOn (WebKeys.assets, fullOptJS in (client, Compile)),
+    stage := { stage dependsOn(WebKeys.assets, fullOptJS in(client, Compile)) }.value,
     routesGenerator := InjectedRoutesGenerator,
     libraryDependencies ++= Seq(
       db.driver,

--- a/project/Dependencies.scala
+++ b/project/Dependencies.scala
@@ -1,12 +1,13 @@
 import sbt._
+import org.scalajs.sbtplugin.ScalaJSCrossVersion
 import org.scalajs.sbtplugin.ScalaJSPlugin.AutoImport._
 
 object Dependencies {
 
   object scalaJs {
     def stubs = "org.scala-js" %% "scalajs-stubs" % scalaJSVersion
-    def angular = "com.greencatsoft" %%%! "scalajs-angular" % "0.8-SNAPSHOT"
-    def jquery = "be.doeraene" %%%! "scalajs-jquery" % "0.9.0"
+    def angular = "com.greencatsoft" % "scalajs-angular" % "0.8-SNAPSHOT" cross ScalaJSCrossVersion.binary
+    def jquery = "be.doeraene" % "scalajs-jquery" % "0.9.0" cross ScalaJSCrossVersion.binary
   }
 
   object js {
@@ -17,7 +18,7 @@ object Dependencies {
   object prickle {
     val version = "1.1.10"
 
-    def js = "com.github.benhutchison" %%%! "prickle" % version
+    def js = "com.github.benhutchison" % "prickle" % version cross ScalaJSCrossVersion.binary
     def jvm = "com.github.benhutchison" %% "prickle" % version
   }
 

--- a/project/build.properties
+++ b/project/build.properties
@@ -1,1 +1,1 @@
-sbt.version=0.13.11
+sbt.version=0.13.17

--- a/project/plugins.sbt
+++ b/project/plugins.sbt
@@ -1,4 +1,6 @@
-addSbtPlugin("org.scala-js" % "sbt-scalajs" % "0.6.16")
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % "0.6.25")
+
+addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "0.6.0")
 
 addSbtPlugin("com.typesafe.sbt" % "sbt-less" % "1.1.1")
 

